File: samples/JavaSamples/src/main/java/com/mku/salmon/samples/main/WebServiceDriveProgram.java
package com.mku.salmon.samples.main;

import com.mku.fs.file.*;
import com.mku.salmon.samples.samples.DriveSample;
import com.mku.salmon.streams.ProviderType;
import com.mku.salmon.streams.AesStream;
import com.mku.salmonfs.drive.AesDrive;

public class WebServiceDriveProgram {
    public static void main(String[] args) throws Exception {
        String wsServicePath = "https://localhost:8443";
        String wsUser = "user";
        String wsPassword = "password";
        String drivePath = "/example_drive_" + System.currentTimeMillis();
        String password = "test123";

		System.out.println("Starting Salmon WebFS Sample");
		System.out.println("make sure your WebFS server is up and running to run this sample");
		
        // enable only if you're testing with an HTTP server
		// In all other cases you should be using an HTTPS server
        // HttpSyncClient.setAllowClearTextTraffic(true);

        AesStream.setAesProviderType(ProviderType.Default);

        IFile[] filesToImport = new File[]{new File("./data/file.txt")};

        IFile dir = new File("./output");
        if (!dir.exists())
            dir.mkdir();
        IFile exportDir = dir.getChild("export");
        if (!exportDir.exists())
            exportDir.mkdir();

        IFile driveDir = new WSFile(drivePath, wsServicePath, new Credentials(wsUser, wsPassword));
        if (!driveDir.exists())
            driveDir.mkdir();

        AesDrive wsDrive = DriveSample.createDrive(driveDir, password);
        wsDrive = DriveSample.openDrive(driveDir, password);
        DriveSample.importFiles(wsDrive, filesToImport);
        DriveSample.listFiles(wsDrive);
        DriveSample.exportFiles(wsDrive, exportDir);
        DriveSample.closeDrive(wsDrive);
    }
}

File: samples/CSharpSamples/CSharpSamples/Main/WebServiceDriveProgram.cs
using Mku.FS.File;
using Mku.Salmon.Samples.Samples;
using Mku.Salmon.Streams;
using Mku.SalmonFS.Drive;
using File = Mku.FS.File.File;
using IFile = Mku.FS.File.IFile;

namespace Mku.Salmon.Samples.Main;

class WebServiceDriveProgram
{
    public static void RunMain(string[] args)
    {
        string wsServicePath = "https://localhost:8443";
        string wsUser = "user";
        string wsPassword = "password";
        string drivePath = "/example_drive_" + Time.Time.CurrentTimeMillis();
        string password = "test123";

		Console.WriteLine("Starting Salmon WebFS Sample");
		Console.WriteLine("make sure your WebFS server is up and running to run this sample");
		
		// enable only if you're testing with an HTTP server
		// In all other cases you should be using an HTTPS server
        // HttpSyncClient.AllowClearTextTraffic = true;
		
        AesStream.AesProviderType = ProviderType.Default;

        IFile[] filesToImport = [new File("./data/file.txt")];

        IFile dir = new File("./output");
        if (!dir.Exists)
            dir.Mkdir();
        IFile exportDir = dir.GetChild("export");
        if (!exportDir.Exists)
            exportDir.Mkdir();

        IFile driveDir = new WSFile(drivePath, wsServicePath, new Credentials(wsUser, wsPassword));
        if (!driveDir.Exists)
            driveDir.Mkdir();

        AesDrive wsDrive = DriveSample.CreateDrive(driveDir, password);
        wsDrive = DriveSample.OpenDrive(driveDir, password);
        DriveSample.ImportFiles(wsDrive, filesToImport);
        DriveSample.ListFiles(wsDrive);
        DriveSample.ExportFiles(wsDrive, exportDir);
        DriveSample.CloseDrive(wsDrive);
    }
}